app/questionnaire/questionnaire_schema.py

```diff
@@ -116,6 +121,55 @@ def placeholder_transform_section_dependencies_by_block(
     def answer_dependencies(self) -> ImmutableDict[str, set[AnswerDependent]]:
         return ImmutableDict(self._answer_dependencies_map)
 
+    @cached_property
+    def min_and_max_map(self) -> Any:
+        return make_immutable(self._min_and_max_map)
+
+    def _create_min_max_map(
+        self,
+        min_max: Literal["minimum", "maximum"],
+        answer_id: str,
+        answers: Iterable[ImmutableDict],
+        default_min_max: int,
+    ) -> None:
+        longest_value_length = 0
+        for answer in answers:
+            value = answer.get(min_max, {}).get("value")
+
+            if isinstance(value, float | int):
+                value_length = len(str(value))
+                decimal_places = answer.get("decimal_places")
+
+                if decimal_places:
+                    value_length = value_length + decimal_places
+
+                longest_value_length = max(longest_value_length, value_length)
+
+            elif isinstance(value, Mapping) and value:
+                if value.get("source") == "answers":
+                    longest_value_length = max(
+                        longest_value_length,
+                        self._min_and_max_map[value["identifier"]][min_max],
+                    )
+
+        self._min_and_max_map[answer_id][min_max] = (
+            longest_value_length or default_min_max
+        )
+
+    def _populate_min_max_for_numeric_answers(self) -> None:
+        for answer_id, answers in self._answers_by_id.items():
+            if answers[0]["type"] in {
+                "Currency",
+                "Duration",
+                "Number",
+                "Percentage",
+                "Unit",
+            }:
```

Comment:
Minor, I think it is worth making this inline set a constant called "NUMERIC_ANSWER_TYPES".